src/test/java/com/squareup/kotlinpoet/TypeSpecTest.kt

```diff
@@ -3568,6 +3569,48 @@ class TypeSpecTest {
       |""".trimMargin())
   }
 
+  // https://github.com/square/kotlinpoet/issues/563
+  @Test fun kdocFormatting() {
+    val typeSpec = TypeSpec.classBuilder("MyType")
+        .addKdoc("This is a thing for stuff.")
+        .addProperty(PropertySpec.builder("first", INT)
+            .initializer("first")
+            .build())
+        .addProperty(PropertySpec.builder("second", INT)
+            .initializer("second")
+            .build())
+        .addProperty(PropertySpec.builder("third", INT)
+            .initializer("third")
+            .build())
+        .primaryConstructor(FunSpec.constructorBuilder()
+            .addKdoc("Construct a thing!")
```

Comment:
Losing this doc is annoying.

Reply:
Should we coalesce it? Not fold when there's constructor docs?

Reply:
There might be a special tag for constructor docs, I'll research